cocos/base/ccUTF8.cpp

```diff
@@ -380,11 +381,27 @@ void StringUTF8::replace(const std::string& newStr)
 
 std::string StringUTF8::getAsCharSequence() const
 {
+    return getAsCharSequence(0, std::numeric_limits<std::size_t>::max());
+}
+
+std::string StringUTF8::getAsCharSequence(std::size_t pos) const
+{
+    return getAsCharSequence(pos, std::numeric_limits<std::size_t>::max());
+}
+
+std::string StringUTF8::getAsCharSequence(std::size_t pos, std::size_t len) const
+{
     std::string charSequence;
+    size_t maxLen = _str.size() - pos;
+    if (len > maxLen)
+    {
+        len = maxLen;
+    }
 
-    for (auto& charUtf8 : _str)
+    len += pos;
```

Comment:
I think it is better to use another variable to make it more clear, such as 
```c++
auto endPos = pos + len;
```

Reply:
maybe `endPos = _str.size();` is better. I modify it